README.md

```diff
@@ -123,69 +49,288 @@ GRANT ALL PRIVILEGES ON DATABASE sample TO manager;
+$ docker rm <image_id>
+...
+```
+
+This now means when you run `docker-compose up` docker will rebuild the image using the new commands in the `Dockerfile`, and then build a new container from that
+
+> *REMEMBER:* To run the commands above when switching back to the other branches on the project
+
+## Tests
+
+```
+$ docker container exec -it wagtail-cms_web_1 python manage.py test
+```
+
+> *IMPORTANT:* When fixing an issue write a test first to prove it's broken. Then, and only then, fix the problem
+
+## Configure
 
-### Running the site
+### Secret Key
 
-#### Running without Docker
+```
+$ python3
+>>> import secrets
+>>> chars = 'abcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*(-_=+)'
+>>> ''.join(secrets.choice(chars) for i in range(50))
+<secret_key>
+>>> exit()
+$ vim docker-compose.yml
+...
+SECRET_KEY: "<secret_key>"
+...
+```
 
-Create a Python 3.6.8 virtual environment and run the following commands from the root directory of the project:
+### Defaults
 
 ```
-pip install -r requirements.txt
-python manage.py migrate
-python manage.py createsuperuser
-python manage.py runserver
+$ cp docker-compose.yml.example docker-compose.yml
+$ vim docker-compose.yml
+...
+DBHOST:               "db"
+DBNAME:               "django"
+DBUSER:               "docker"
+DBPASSWORD:           "docker"
+...
+
+...
+MONGODB_HOST:         "mongo"
+MONGODB_USERNAME:     "mongodb"
+MONGODB_PASSWORD:     "mongodb"
+...
+SEARCHAPIHOST:        "..."
+...
+# docker-compose up
 ```
 
+> If you are going to be importing new data to the fixtures remember to set DATASETAPIHOST, DATASETAPIKEY, AZURECOSMOSDBURI and AZURECOSMOSDBKEY
 
-#### Running with Docker
+## Docker
+
+You need to have [Docker](https://docs.docker.com/install/) installed to be able to see changes in the code, run tests and update local data.
 
 ```
-docker-compose build
-docker-compose up
+$ docker-compose up
 ```
 
-The first command builds the docker image.
-The second command starts the docker container, running on port 8000.
+> Make sure to run this command in a separate window so you have somewhere to run other commands from
+
+> This command will download all the required images (Python, PostgreSQL, CosmosDB) and start them for you all preconfigured
 
-After the docker image is running for the first time, connect to it and create an admin login as follows:
+# Database (PostgreSQL)
+
+## Populating
 
-(From the route directory)
 ```
-./bin/manage createsuperuser
+$ docker container exec -it wagtail-cms_web_1 python manage.py populate_cms
 ```
 
-##### Useful commands
+### Updating Postgres Data *(CMS/fixtures/postgres.json)*
 
-There are commands setup in the bin directory of the project, that allow easy use of common commands inside the docker container.
+```
+$ docker container exec -it wagtail-cms_web_1 python manage.py update_cms_fixture --db_host ... --db_name ... --db_user ... --db_password ... --db_port ...
+```
```

Comment:
Correct. Also it will not run if the `docker-compose.yml` values for the database do not point at the local database (https://github.com/office-for-students/wagtail-CMS/blob/rob-fixes/content/management/commands/update_cms_fixture.py#L43) in docker. This means you can't accidentally run it and start truncating and putting things in remote databases.